File: app/src/main/java/com/example/java/android1/java_android_notes/service/NotesAdapter.java
package com.example.java.android1.java_android_notes.service;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.java.android1.java_android_notes.R;
import com.example.java.android1.java_android_notes.data.DataNote;
import com.example.java.android1.java_android_notes.data.DataNoteSource;
import com.example.java.android1.java_android_notes.listeners.OnItemClickListener;
import com.example.java.android1.java_android_notes.listeners.OnItemLongClickListener;
import com.example.java.android1.java_android_notes.ui.ListOfNotesFragment;

public class NotesAdapter extends RecyclerView.Adapter<ViewHolder> {

    private final LayoutInflater inflater;
    private final DataNoteSource mDataNoteSource;
    private OnItemClickListener mOnItemClickListener;
    private final ListOfNotesFragment mNotesFragment;
    private OnItemLongClickListener mOnItemLongClickListener;

    public NotesAdapter(ListOfNotesFragment notesFragment, DataNoteSource dataNoteSource) {
        this.mNotesFragment = notesFragment;
        this.mDataNoteSource = dataNoteSource;
        this.inflater = mNotesFragment.getLayoutInflater();
    }

    public void setOnItemClickListener(OnItemClickListener onItemClickListener) {
        this.mOnItemClickListener = onItemClickListener;
    }

    public void setOnItemLongClickListener(OnItemLongClickListener onItemLongClickListener) {
        this.mOnItemLongClickListener = onItemLongClickListener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = inflater.inflate(R.layout.note_item, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        DataNote dataNote = mDataNoteSource.getItem(position);
        holder.itemView.setOnClickListener((view) -> {
            if (mOnItemClickListener != null) {
                mOnItemClickListener.onItemClickListener(view, position);
            }
        });
        holder.fillCard(mNotesFragment, dataNote, mOnItemLongClickListener);
        holder.setFavoriteNote(mDataNoteSource, position);
    }

    @Override
    public int getItemCount() {
        return mDataNoteSource.getDataNoteCount();
    }
}

File: app/src/main/java/com/example/java/android1/java_android_notes/service/NotesAdapter.kt
package com.example.java.android1.java_android_notes.service

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.example.java.android1.java_android_notes.data.DataNoteSource
import com.example.java.android1.java_android_notes.databinding.NoteItemBinding
import com.example.java.android1.java_android_notes.ui.ListOfNotesFragment

class NotesAdapter(
    private val notesFragment: ListOfNotesFragment,
    private val dataNoteSource: DataNoteSource?,
    private val onItemClickListener: ((position: Int) -> Unit)?,
    private val onItemLongClickListener: ((position: Int) -> Unit)?
) : RecyclerView.Adapter<ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder = ViewHolder(
        NoteItemBinding.inflate(LayoutInflater.from(parent.context), parent, false)
    )

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val dataNote = dataNoteSource?.getItem(position)
        holder.itemView.setOnClickListener {
            onItemClickListener?.invoke(position)
        }
        holder.bindCard(notesFragment, dataNote, onItemLongClickListener)
        dataNoteSource?.let { holder.setFavoriteNote(it, position) }
    }

    override fun getItemCount(): Int = dataNoteSource?.dataNoteCount ?: 0
}